packages/editor/src/bindings/post-meta.js

```diff
@@ -77,19 +77,64 @@ export default {
 		return true;
 	},
 	getFieldsList( { registry, context } ) {
-		const metaFields = registry
-			.select( coreDataStore )
-			.getEditedEntityRecord(
+		let metaFields = {};
+		const {
+			type,
+			is_custom: isCustom,
+			slug,
+		} = registry.select( editorStore ).getCurrentPost();
+		const { getRegisteredPostMeta, getPostTypes, getEditedEntityRecord } =
+			registry.select( coreDataStore );
+
+		// Inherit the postType from the slug if it is a template.
+		if ( ! context?.postType && type === 'wp_template' ) {
+			// Get the 'kind' from the start of the slug.
+			// Use 'post' as the default.
+			let postType = 'post';
+			const isGlobalTemplate = isCustom || slug === 'index';
+			if ( ! isGlobalTemplate ) {
+				const [ kind ] = slug.split( '-' );
+				if ( kind === 'page' ) {
+					postType = 'page';
+				} else if ( kind === 'single' ) {
+					const postTypes =
+						getPostTypes( { per_page: -1 } )?.map(
+							( entity ) => entity.slug
+						) || [];
+
+					// Infer the post type from the slug.
+					// TODO: Review, as it may not have a post type. http://localhost:8888/wp-admin/site-editor.php?canvas=edit
+					const match = slug.match(
+						`^single-(${ postTypes.join( '|' ) })(?:-.+)?$`
+					);
+					postType = match ? match[ 1 ] : 'post';
```

Comment:
I think right now this PR handles the archives pretty well, both custom and general. Once we merge these will be in a "quite ready" state.

https://github.com/WordPress/gutenberg/pull/65116
https://github.com/WordPress/gutenberg/pull/65099
https://github.com/WordPress/gutenberg/pull/65062

![Screenshot 2024-09-06 at 16 48 43](https://github.com/user-attachments/assets/d8cbb2a0-91c8-4e86-a1e8-708ec9319a11)

There is still time for discussion and testing, and this usually don't happen until the first Beta. It can be reverted or punted for 6.9, but there is still a couple of months left for doing that.